### PR TITLE
Fix/SOC-6041: Distinguish between ip address and section numbering

### PR DESCRIPTION
**Pull request by hbenali**
Editing Regex Expression in order to distinguish between the IP address and section numbering. In order to parse an IP address as a link, the user has to write the protocol ((http|ftp)(s)://) before the ip.

### CODE REVIEW

File: component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPlugin.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
   private static final Pattern URL_PATTERN = Pattern
       .compile("(?i)" +
       "(" +
-        "((?:(?:ht|f)tp(?:s?)\\:\\/\\/)?" +                                                       // protolcol
+        "((?:(?:ht|f)tp(?:s?)\\:\\/\\/)" +                                                       // protolcol
```
Comment (thomasdelhomenie) [inline component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPlugin.java]: It means urls without the protocol won't be recognized as links, right ? If so, it is a regression.

Comment (hbenali) [reply component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPlugin.java]: it is only for ip addresses.

Comment (hbenali) [reply component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPlugin.java]: For the domain name. i kept the protocol adding optional

Comment (thomasdelhomenie) [reply component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPlugin.java]: > it is only for ip addresses

What do you mean ? The regex is used to detect all urls. If we don't keep the protocol optional, it means that text like "www.exoplatform.com" won't be recognized as link anymore, no ?

Comment (hbenali) [reply component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPlugin.java]: After discussing with Hamadi. he considers the old behavior as a bug (Also he commented on Jira Ticket). I have deleted the pattern of IP Address.

Comment (thomasdelhomenie) [reply component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPlugin.java]: ok, sound good. Can you add unit test to validate it does transform an IP address in link anymore please ?

Comment (hbenali) [reply component/common/src/main/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPlugin.java]: Done

File: component/common/src/test/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPluginTest.java [hunk 2]
```diff
@@ -31,6 +31,14 @@
   public void testURLConverterFilterPlugin() {
     Filter urlConverterFilter = new URLConverterFilterPlugin(0);
 
+    assertEquals("http://192.168.1.1/", urlConverterFilter.doFilter("http://192.168.1.1/"));
```
Comment (thomasdelhomenie) [inline component/common/src/test/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPluginTest.java]: Is it expected to not transform a text starting with "http" in a link ? I think it should. For example I may want to give an internal URL to a colleague using the IP. As long as it starts with http|https|ftp it should be a link IMO.

Comment (hbenali) [reply component/common/src/test/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPluginTest.java]: i have fixed it. The user should add the protocol before the ip address inorder to transform it as a link.

Comment (thomasdelhomenie) [reply component/common/src/test/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPluginTest.java]: Yes indeed, but we come back to my initial comment about text like "www.exoplatform.com" not being recognized as link :)
I think we should have the following behavior:
* http://www.exoplatform.com : link
* www.exoplatform.com : link
* http://192.168.1.19 : link
* 192.168.1.19 : not a link

Comment (thomasdelhomenie) [reply component/common/src/test/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPluginTest.java]: As you can see in my previous comment, this is how github works :)

Comment (hbenali) [reply component/common/src/test/java/org/exoplatform/social/common/xmlprocessor/filters/URLConverterFilterPluginTest.java]: it works fine now :)  I have added a new unit test

**Pull request merged**